isyfact-standards-doc/src/docs/antora/modules/isy-security/pages/konzept/inhalt.adoc

```diff
@@ -143,53 +146,83 @@ image::konzept/security-schnittstellen.dn.svg[align="center"]
 
 Das Interface `Security` ist der zentrale Einstiegspunkt in den Baustein `isy-security`.
 Bei seiner Instanziierung wird die anwendungsspezifische Rollen-Rechte-Datei eingelesen und damit der `Authentifizierungsmanager` sowie der `Berechtigungsmanager` erzeugt.
-Das Interface `Security` stellt der Fachanwendung die im weiteren beschriebenen `Authentifizierungsmanager` und `Berechtigungsmanager` der Fachanwendung zur Verfügung.
-Außerdem kann die Fachanwendung über die Methode `getAlleRollen` eine Liste von allen im System hinterlegten Rollen erhalten.
+Das Interface `Security` stellt der Geschäftsanwendung die im Weiteren beschriebenen `Authentifizierungsmanager` und `Berechtigungsmanager` der Fachanwendung zur Verfügung.
+Außerdem kann die Geschäftsanwendung über die Methode `getAlleRollen` eine Liste von allen im System hinterlegten Rollen erhalten.
+
+Der `Authentifizierungsmanager` übernimmt die programmatische Authentifizierung und Autorisierung interner und externer Anwendungen durch das IAM für den Zugang zu IsyFact-Nachbaranwendungen.
+Dafür sind folgende Szenarien zu unterscheiden, für die es jeweils eine eigene Servicemethode gibt:
 
-Der `Authentifizierungsmanager` übernimmt die Authentifizierung von technischen Nutzern am IAM-System.
-Mit der Methode `authentifiziereClient` wird die OAuth2.0-konforme Authentifizierung eines Client mittels Client-ID u. Client-Secret unterstützt und mit der Methode `authentifiziereSystem` der alternative Weg mittels (techn.) Nutzername u. Passwort.
-Bei erfolgreicher Authentifizierung werden die vom IAM-System zurückgegebenen Daten (Access-Token, Rollen des Anwenders, ...) im `Spring SecurityContext` abgelegt.
-Dort werden auch die den aus den Rollen resultierenden Rechte des Anwenders gespeichert.
+==== Authentifizierung einer internen Anwendung (Batch, Task, …) als eigener Schritt (direkt/intern):
+
+Interne Anwendungen sind als Confidential Client eingestuft und verwahren ihre Authentifizierungsdaten (entweder Client-ID u. Client-Secret oder Nutzername u. Passwort) in ihrer Konfiguration (mit einer ApplicationID als Schlüssel).
+Die Methode `authentifiziere(applicationID)` im Authentifizierungsmanager liest diese Authentifizierungsdaten aus der betrieblichen Konfiguration und führt damit eine OAuth2.0-konforme Authentifizierung und Autorisierung der Anwendung gegenüber dem IAM-System durch.
+Dabei wird innerhalb der Methode `authentifiziere(applicationID)` anhand der property `authorization-grant-type` der passende Authentication-Flow ausgewählt:
+Der Wert 'client-credentials' bedeutet, den _Credential_-Flow xref:client-credential-flow[] und damit die Methode `authentifiziereClient` zu verwenden, der Wert 'password'  steht für den _Credential_-Flow xref:resource-owner-password-credential-flow[] für den die Methode `authentifiziereSystem` zu verwenden ist - jeweils mit der ApplicationID und den Authentifizierungsdaten der internen Anwendung. +
```

Comment:
Mit Aufruf der Methode `autentifiziere(applicationID)` erfolgt intern die Entscheidung des entsprechenden credential-flows, aber danach erfolgt kein Aufruf der Flow spezifischen Methoden des AuthentifizierungsManagers `authentifiziereSystem` bzw `authentifiziereClient`.

Die interne Flow spezifische Abarbeitung der Authentifizierung erfolgt mit den Flow spezifischen `ClientCredentialsAuthenticationProvider`/`ClientCredentialsAuthenticationToken` bzw. `PasswordAuthenticationProvider`/`PasswordAuthenticationToken`.

Die beiden Flow spezifischen Methoden des `BerechtigungsManagers` werden nur von Anwendungen aufgerufen, die vorher schon die Entscheidung des zu verwendenden CredentialFlows getroffen haben, wie beispielsweise ein SGW.

Reply:
Zum einen dachte ich, dass in `authentifiziere` nach Feststellung des zu verwendenden credential flow die weitere Verarbeitung die selbe ist, wie in `authentifiziereSystem `bzw `authentifiziereClient`. Dem ist anscheinend nicht so.
Zum anderen verlangt der resource-owner-password-credential-flow zusätzlich noch credentials für einen confidential client.  
Ich wollte festlegen, dass dafür auch die (selben) credentials des internen Systems verwendet werden sollen. Das ist aber whl noch zu klären bzw. Implementierungsdetails, die nicht ins Konzept gehören. 

=> Erwähnung des Aufrufes von `authentifiziereSystem `bzw `authentifiziereClient` und der Verwendung des internen Systems als confidential client gestrichen.